### PR TITLE
Cu 86by51hf7 create form unique to interaction

### PR DESCRIPTION
**Pull request by NyashaMuusha**
unique to interaction

### CODE REVIEW

**Comment by rcrichton** [comment]
Task linked: [CU-86by51hf7 Create Form: Unique to Interaction](https://app.clickup.com/t/86by51hf7)

**Comment by rcrichton** [comment]
Task linked: [CU-86by1w64p JeMPI UI Configuration](https://app.clickup.com/t/86by1w64p)

File: JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToInteraction/UniqueToInteraction.tsx [hunk 1]
```diff
@@ -104,48 +57,73 @@ const UniqueToInteraction = () => {
...
       field: 'index',
       headerName: 'Index',
       type: 'string',
       width: 300,
       align: 'center',
       headerAlign: 'center',
-      editable: false
+      editable: false,
+      valueGetter: params => {
+        const indexGoldenRecord = params.row.indexGoldenRecord
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToInteraction/UniqueToInteraction.tsx:119]: @NyashaMuusha Can you please refactor and extract this method to a hepler that performs this logic for all index fields in tabs

**Review by MatthewErispe — CHANGES_REQUESTED** [review]

File: JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToInteraction/UniqueToInteraction.tsx [hunk 2]
```diff
@@ -203,23 +184,40 @@ const UniqueToInteraction = () => {
...
+          slotProps={{
+            toolbar: { setRows, setRowModesModel }
+          }}
+        />
+      )}
     </Box>
   )
 }
 
+function EditToolbar(props: EditToolbarProps) {
```
Comment (MatthewErispe) [inline JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToInteraction/UniqueToInteraction.tsx:208]: @NyashaMuusha  Requested refactor in previous changes #235 . Please use the helper function once completed

**Review by MatthewErispe — APPROVED** [review]
@NyashaMuusha LGTM

**Pull request merged**